sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -691,7 +692,8 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
   def warehousePath: String = {
-    getConf(WAREHOUSE_PATH).replace("${system:user.dir}", System.getProperty("user.dir"))
+    new Path(getConf(WAREHOUSE_PATH).replace("${system:user.dir}",
+      System.getProperty("user.dir"))).toUri.toString
```

Comment:
It is equivalent to `new Path`. This does not add a leading slash, and we need to check if it is still needed with the changes we introduced.
```
scala> new Path(new Path("c:\\foo path").toUri).toString
res1: String = c:/foo path
scala> new Path("c:\\foo path")
res2: org.apache.hadoop.fs.Path = c:/foo path
scala> new Path("c:\\foo path").toUri
res3: java.net.URI = /c:/foo%20path
```
Basically, it retur